weDevsOfficial/dokan
# enhance: Background processor for rebuilding variable products author IDs

### All Submissions:






### Changes proposed in this Pull Request:



 * Integrated background processor for rebuilding variable products author IDs
 * This PR is associated with `Dokan Pro` PR: https://github.com/weDevsOfficial/dokan-pro/pull/2143

### How to test the changes in this Pull Request:

1. Change IDs of some variable products from the posts table's `post_author` column, where `post_type` is `product_variation`.
2. Run upgrader from `Admin Dashboard`.
3. After running the upgrader, previously changed IDs should be restored as per the parent variation product.



### Changelog entry

> enhance: Integrated background processor for rebuilding variable products author IDs


### PR Self Review Checklist:

* Code is not following code style guidelines
* Bad naming: make sure you would understand your code if you read it a few months from now.
* KISS: Keep it simple, Sweetie (not stupid!).
* DRY: Don't Repeat Yourself.
* Code that is not readable: too many nested 'if's are a bad sign.
* Performance issues
* Complicated constructions that need refactoring or comments: code should almost always be self-explanatory.
* Grammar errors.


### FOR PR REVIEWER ONLY:

> As a reviewer, your feedback should be focused on the idea, not the person. Seek to understand, be respectful, and focus on constructive dialog.

> As a contributor, your responsibility is to learn from suggestions and iterate your pull request should it be needed based on feedback. Seek to collaborate and produce the best possible contribution to the greater whole.
